### ISSUE TITLE
Proxy support regressed

### ISSUE THREAD

**Issue by springmeyer** [mapbox/node-pre-gyp#400]
It looks like the move to needle regressed proxy support for some users: https://github.com/mapbox/node-sqlite3/issues/1010.

I don't have the time or ability to debug this. Anyone familiar with proxies have the bandwidth to take a look? Seems like the options would be to use needles support for passing a custom agent or revert back to request.

**Comment by springmeyer** [comment]
I've released v0.10.3 which will now use `request` if you install it separetly. Can anyone experience proxy support try doing:

```
npm install node-pre-gyp@0.10.3 request --save
```

And then see if proxy support works for their module that uses node-pre-gyp?

**Comment by nschonni** [comment]
Request 2.87.0 fixed those hoek/hawk issues and keeps the 4> comp again so it might be safe to just drop needle and pin request again.

**Comment by springmeyer** [comment]
@nschonni yes, going back to request feels safest. I did do #404, which will use request if found, but I'm feeling like that is going to be confusing and a better solution would be just to use request over needle. Thoughts?

**Comment by nschonni** [comment]
My 2 cents would be to go back to `request`. node-gyp also has an open PR to pin to 2.87 too

**Comment by mapsam** [comment]
@springmeyer also in favor of moving back to request 👍- especially since we no longer recommend bundling node-pre-gyp when releasing downstream modules, the size of the full library will be less of an issue.

**Comment by springmeyer** [comment]
👍 to moving back to request then. Feel free to run that out @mapsam.

**Comment by redbaron** [comment]
@springmeyer

thanks for your comment, installing request made a huge difference!

**Comment by juliomelo** [comment]
```
npm install node-pre-gyp@0.13.0 request --save
```

This solved my problem. Here we have a Squid HTTP proxy using DansGuardian and using node-pre-gyp 0.13.0, the proxy reject with HTTP 400 and the following payload:

```html
<HTML><HEAD><TITLE>DansGuardian - 400 Bad Request</TITLE></HEAD><BODY><H1>DansGuardian - 400 Bad Request</H1> A URL requerida está mal formada.</BODY></HTML>
```

The HTTP messsage was:

```
GET https://github.com/node-gfx/node-canvas-prebuilt/releases/download/v2.5.0/canvas-v2.5.0-node-v64-linux-glibc-x64.tar.gz HTTP/1.1
accept: */*
user-agent: node-pre-gyp (v0.13.0, node v10.16.3)
host: github.com
Connection: close
```

The problem is that it is using GET method over a HTTP proxy to retrieve a HTTPS url. It should use CONNECT method instead.

I do not know details of node-pre-gyp implementation, neither of that "request" package, as suggested by @springmeyer, but it solves the problem for me.
